pkg/repository/git.go

```diff
@@ -33,29 +34,47 @@ func newGitRepoController() gitRepoController {
 	}
 }
 
+type cleanStatus struct {
+	IsClean bool
+	Err     error
+}
+
 // IsClean checks if the repository have changes over the commit
 func (r gitRepoController) isClean(ctx context.Context) (bool, error) {
 	ctxWithTimeout, cancel := context.WithTimeout(ctx, 1*time.Second)
 	defer cancel()
-	select {
```

Comment:
the previous timeout was ineffective because all function calls below did not actually use this context and they were not in a coroutine either so they would just execute and the timeout would kick in but without any effect.